src/puptoo/process/profile.py

```diff
@@ -457,18 +457,18 @@ def system_profile(
             catch_error("yum_repos_d", e)
             raise
 
-    if dnf_modules:
+    if dnf_module_list:
         try:
-            modules = []
-            for module in dnf_modules:
-                for module_name in module.sections():
-                    modules.append(
-                        {
-                            "name": module_name,
-                            "stream": module.get(module_name, "stream"),
-                        }
-                    )
-            profile["dnf_modules"] = sorted(modules, key=lambda k: k["name"])
+            modules = {}
+            for _, module in dnf_module_list.items():
+                if "[e]" in module.stream or ("[d]" in module.stream and "[x]" not in module.stream):
+                    # use only active modules (enabled or default)
+                    stream = module.stream.replace("[d]", "").replace("[e]", "").strip()
```

Comment:
Maybe cleanup all special bracket parts (to be resilient for future enhancements).